src/routes/orphanedData/index.js

```diff
@@ -1,143 +1,65 @@
-import React from 'react'
+import React, { useEffect } from 'react'
 import PropTypes from 'prop-types'
-import { routerRedux } from 'dva/router'
+import { Tabs } from 'antd'
 import { connect } from 'dva'
-import { Row, Col, Select } from 'antd'
-import { Filter } from '../../components/index'
-import OrphanedDataList from './orphanedDataList'
-import OrphanedDataBulkActions from './orphanedDataBulkActions'
-import queryString from 'query-string'
-import C from '../../utils/constants'
-
-const Option = Select.Option
-
-class OrphanedData extends React.Component {
-  constructor(props) {
-    super(props)
-    this.state = {
-      height: 300,
-      selectedRows: [],
-    }
-  }
-
-  componentDidMount() {
-    this.onResize()
-    window.addEventListener('resize', this.onResize)
-  }
-
-  componentWillUnmount() {
-    window.removeEventListener('resize', this.onResize)
-  }
-
-  onResize = () => {
-    const height = document.getElementById('orphanedDataTable').offsetHeight - C.ContainerMarginHeight
-    this.setState({
-      height,
-    })
-  }
-
-  onResourceTypeChange = () => {
-    // to do -> Support other resource type
-    return null
-  }
+import { routerRedux } from 'dva/router'
+import ReplicaOrphans from './ReplicaOrphans'
+import InstanceOrphans from './InstanceOrphans'
+import styles from './index.less'
+import { ORPHAN_TYPES } from '../../utils/orphanedData'
 
-  render() {
-    const { loading, dispatch, location, orphanedData } = this.props
-    const data = orphanedData.data
-    const me = this
+function OrphanedData({ dispatch, location }) {
+  const { pathname, hash, search } = location
 
-    const orphanedDataBulkActionsProps = {
-      selectedRows: this.state.selectedRows,
-      deleteorphanedData(record) {
-        if (record?.length > 0) {
-          dispatch({
-            type: 'orphanedData/bulkDelete',
-            payload: record.map(item => { return { name: item.name } }),
-            callback: () => {
-              me.setState({
-                ...me.state,
-                selectedRows: [],
-              })
-            },
-          })
-        }
-      },
-    }
+  const tabs = [
+    {
+      key: ORPHAN_TYPES.REPLICA,
+      label: 'Replica Orphans',
+      content: <ReplicaOrphans location={location} />,
+    },
+    {
+      key: 'instance',
+      label: 'Instance Orphans',
+      content: <InstanceOrphans location={location} />,
+    },
+  ]
```

Comment:
I think using tabs to switch between orphan types could offer better UX than separate pages, but I'm open to any suggestions.

Reply:
Great idea, tabs are more convenient then pages.